back/benchmarks/Main.hs

```diff
@@ -0,0 +1,49 @@
+-- | Module contains all stuff to migrate from AcidState to Postgres.
+module Main
+       (
+         main
+       ) where
+
+import Imports
+
+import Gauge
+import Hasql.Transaction.Sessions (Mode (..))
+
+import Guide.Database.Queries.Update
+import Guide.Database.Queries.Select
+import Guide.Types.Core
+import Guide.Database.Connection
+
+
+main :: IO ()
+main = do
+    conn <- connect
+    defaultMain [databaseBenchmark conn]
+  where
+    databaseBenchmark conn =
+      bgroup
+        "Database"
+        [ bench "select" $ nfIO $
+            runTransactionExceptT conn Read $ selectCategory "category1111"
+        , bench "updete" $ nfIO $
+            runTransactionExceptT conn Write $ updateCategory "category1111" update
+        ]
+    update :: Category -> Category
+    update = _categoryTitle .~ "title10"
```

Comment:
After the first application of `update` the title will be `"title10"`, so `update` will be equivalent to `id` and `updateCategory` will not do any writes. You should do something that would change the category _every_ time, e.g. `_categoryTitle <>~ "+"`, and then re-run the benchmarks.

Reply:
ok. then speed is
![code](https://user-images.githubusercontent.com/28530747/64076790-ac7c1900-cce2-11e9-9faf-6cb25cd873d2.png)